sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/models/PathProperties.java

```diff
@@ -132,6 +133,54 @@ public PathProperties(final OffsetDateTime creationTime, final OffsetDateTime la
         final Boolean isIncrementalCopy, final AccessTier accessTier, final ArchiveStatus archiveStatus,
         final String encryptionKeySha256, final OffsetDateTime accessTierChangeTime,
         final Map<String, String> metadata, final OffsetDateTime expiresOn) {
+        this(creationTime, lastModified, eTag, fileSize, contentType, contentMd5, contentEncoding, contentDisposition,
+            contentLanguage, cacheControl, leaseStatus, leaseState, leaseDuration, copyId, copyStatus, copySource,
+            copyProgress, copyCompletionTime, copyStatusDescription, isServerEncrypted, isIncrementalCopy, accessTier,
+            archiveStatus, encryptionKeySha256, accessTierChangeTime, metadata, expiresOn, null);
+    }
+
+    /**
+     * Constructs a {@link PathProperties}.
+     *
+     * @param creationTime Creation time of the file.
+     * @param lastModified Datetime when the file was last modified.
+     * @param eTag ETag of the file.
+     * @param fileSize Size of the file.
+     * @param contentType Content type specified for the file.
+     * @param contentMd5 Content MD5 specified for the file.
+     * @param contentEncoding Content encoding specified for the file.
+     * @param contentDisposition Content disposition specified for the file.
+     * @param contentLanguage Content language specified for the file.
+     * @param cacheControl Cache control specified for the file.
+     * @param leaseStatus Status of the lease on the file.
+     * @param leaseState State of the lease on the file.
+     * @param leaseDuration Type of lease on the file.
+     * @param copyId Identifier of the last copy operation performed on the file.
+     * @param copyStatus Status of the last copy operation performed on the file.
+     * @param copySource Source of the last copy operation performed on the file.
+     * @param copyProgress Progress of the last copy operation performed on the file.
+     * @param copyCompletionTime Datetime when the last copy operation on the file completed.
+     * @param copyStatusDescription Description of the last copy operation on the file.
+     * @param isServerEncrypted Flag indicating if the file's content is encrypted on the server.
+     * @param isIncrementalCopy Flag indicating if the file was incrementally copied.
+     * @param accessTier Access tier of the file.
+     * @param archiveStatus Archive status of the file.
+     * @param encryptionKeySha256 SHA256 of the customer provided encryption key used to encrypt the file on the server.
+     * @param accessTierChangeTime Datetime when the access tier of the file last changed.
+     * @param metadata Metadata associated with the file.
+     * pass {@code null}.
+     * @param expiresOn the time when the path is going to expire.
+     * @param encryptionScope The path's default encryption scope to encrypt data with.
+     */
+    public PathProperties(final OffsetDateTime creationTime, final OffsetDateTime lastModified, final String eTag,
```

Comment:
Should `FileSystemProperties`, `PathItem`, and this (`PathProperties`) begin following the new pattern being used in [`BlobProperties`](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/models/BlobProperties.java#L25) where there is an internal interface that manages all of the properties where each time new properties are added it's just exposing getters instead of needing new constructors like this.